NitroxModel/DataStructures/Optional.cs

```diff
@@ -150,14 +150,21 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
         }
     }
 
+
     public struct OptionalEmpty
     {
+        [Obsolete("Use Optional.Empty instead")]
```

Comment:
@Measurity

Reply:
A struct constructor can't be non-public, you can always do `default(MyStruct)`. It's because a struct is just a sequence of bits and you can always initialize them to all 0, unlike the concept of a `class`.